src/frontends/tensorflow_common/src/op/unsorted_segment_sum.cpp

```diff
@@ -40,16 +41,32 @@ OutputVector translate_unsorted_segment_sum_op(const NodeContext& node) {
     auto const_zero_i64 = make_shared<v0::Constant>(element::i64, Shape{}, 0);
     auto const_one_i64 = make_shared<v0::Constant>(element::i64, Shape{1}, 1);
     auto const_one_i64_scalar = make_shared<v0::Constant>(element::i64, Shape{}, 1);
+    auto const_minus_one_i64 = make_shared<v0::Constant>(element::i64, Shape{1}, -1);
     auto data_const_zero = create_same_type_const_scalar<float>(data, 0.0f);
 
+    Output<Node> data_shape = make_shared<v3::ShapeOf>(data, element::i64);
+
+    // adjust data and segment_ids inputs for ND indices
+    // to make indices 1D tensor
+    // for example, data shape = [2, 3, 4] and segment_ids shape - [2, 3]
+    // so they need adjustment to new data shape [6, 4] and segment_ids shape [6]
+    auto segment_ids_rank = compute_subgraph_scalar_rank(segment_ids, element::i64, false);
+    // 1. segment_ids needs to be flatten
+    segment_ids = make_shared<v1::Reshape>(segment_ids, const_minus_one_i64, false);
+    // 2. flatten first (segment_ids_rank - 1) dimensions into one dimension
+    auto stop_const = make_shared<v0::Constant>(element::i64, Shape{1}, numeric_limits<int64_t>::max());
+    auto step_const = make_shared<v0::Constant>(element::i64, Shape{1}, 1);
+    Output<Node> slice_shape = make_shared<v8::Slice>(data_shape, segment_ids_rank, stop_const, step_const);
+    auto new_data_shape = make_shared<v0::Concat>(OutputVector{const_minus_one_i64, slice_shape}, 0);
+    data = make_shared<v1::Reshape>(data, new_data_shape, false);
+
     // segment ids can be negative for which the resulted data will be zeroed
     // so it needs to introduce default slice of zeros in the data
     // 1. create default slice that will be used for negative segment ids
-    auto data_shape = make_shared<v3::ShapeOf>(data, element::i64);
-    auto slice_shape = get_data_slice(data_shape, 1, numeric_limits<int>::max(), 1);
     slice_shape = make_shared<v0::Concat>(OutputVector{const_one_i64, slice_shape}, 0);
     auto default_slice = make_shared<v3::Broadcast>(data_const_zero, slice_shape);
     // 2. update data with the default slice
+    data_shape = make_shared<v3::ShapeOf>(data, element::i64);
```

Comment:
(To consider) Looks like `new_data_shape` could be reused in L74 instead of `data_shape` recalculation by `ShapeOf` 
https://github.com/openvinotoolkit/openvino/blob/1d6d62bde86419deb4136c83c3c95e5a7770fd58/src/frontends/tensorflow_common/src/op/unsorted_segment_sum.cpp#L74

Reply:
`new_data_shape` will not work because it contains `-1` for dimension 0

Reply:
Ok, got it.